### PR TITLE
[Develop] created new component Spin

### PR DESCRIPTION
**Pull request by naimdanialjohar**
# Description

created new component Spin

## Type of change

- [ ] New feature (non-breaking change which adds functionality)

# Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] My changes generate no new warnings

### CODE REVIEW

**Review by Kelvin-GohKengBoon — COMMENTED** [review]
LGTM, Let's wait for @AladdinNullptr's review.

File: src/Spin/index.tsx [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+import './style.css'
+
+export type SpinProps = AntSpinProps
+
+const iconSize = (size: SpinSize | number) => {
+  if (size === 'small') return 14
+  else if (size === 'default') return 20
+  else if (size === 'large') return 32
+  else return size
+}
```
Comment (AladdinNullptr) [inline src/Spin/index.tsx:15]: this can be set as an object { small: 14, default: 20, large: 32}

File: src/Spin/index.tsx [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+export const SpinIcon: React.FC<{ size: SpinSize | number }> = props => {
+  const { size = 'default' } = props
+  return <LoadingOutlined style={{ fontSize: iconSize(size) }} />
+}
+
+const Spin: React.FC<SpinProps> = props => {
+  const { indicator, size = 'default', ...restProps } = props
+  return <AntSpin indicator={<SpinIcon size={size} /> || indicator} {...restProps} />
```
Comment (AladdinNullptr) [inline src/Spin/index.tsx:24]: indicator can be set as a default prop like mentioned below

File: src/Spin/style.css [hunk 3]
```diff
@@ -0,0 +1,10 @@
+@import 'antd/lib/style/default.css';
+@import 'antd/lib/spin/style/index.css';
+
+.ant-spin {
```
[the remark this answers is not in the document]
Comment (naimdanialjohar) [reply src/Spin/style.css:4]: i applied tailwindcss class to the `LoadingOutlined` icon in the default prop, the `color` get overridden by other css classes. so i still need to apply it in the class

image

File: src/Spin/index.tsx (hunk 2 above)
Comment (naimdanialjohar) [reply src/Spin/index.tsx:24]: if we set indicator as default prop, when we do want to change the sizes, we still need to declare `indicator` and pass the size as follows:
```
<Spin indicator={<SpinIcon size='large'} />
```
current code allows us to change the indicator size only by passing the `size` prop, and still preserve `SpinIcon` as the default indicator, without declaring it in the `defaultProps`

File: src/Spin/style.css (hunk 3 above)
[the remark this answers is not in the document]
Comment (AladdinNullptr) [reply src/Spin/style.css:4]: You can add a class `ui-default-spin-icon` to `LoadingOutlined` and modify that class

File: src/Spin/index.tsx [hunk 4]
```diff
@@ -0,0 +1,33 @@
+import * as React from 'react'
+import { LoadingOutlined } from '@ant-design/icons'
+import { Spin as AntSpin, SpinProps as AntSpinProps } from 'antd'
+import { SpinSize } from 'antd/lib/spin'
+
+import './style.css'
+
+export type SpinProps = AntSpinProps
+const SpinSizeNumbers = {
```
Comment (AladdinNullptr) [inline src/Spin/index.tsx:9]: export

File: src/Spin/index.tsx [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+  default: 20,
+  large: 32
+}
+
+const iconSize = (size: SpinSize | number) => {
+  if (size === 'small') return SpinSizeNumbers.small
+  else if (size === 'default') return SpinSizeNumbers.default
+  else if (size === 'large') return SpinSizeNumbers.large
+  else return size
+}
```
Comment (AladdinNullptr) [inline src/Spin/index.tsx:20]: remove this function

**Review by AladdinNullptr — APPROVED** [review]

**Pull request merged**